### PR TITLE
Use SQL for Create, Update, and Delete groups

### PR DESCRIPTION
**Pull request by dnephin**
## Summary

Updates the data query functions for `CreateGroup`, `AddUserstoGroup`, `RemoveUsersFromGroup`, and `DeleteGroup` to use SQL.

Related to #2415

### CODE REVIEW

File: internal/server/models/group.go [hunk 1]
```diff
@@ -13,9 +13,7 @@ type Group struct {
 	CreatedBy         uid.ID
 	CreatedByProvider uid.ID
 
-	Identities []Identity `gorm:"many2many:identities_groups"`
```
Comment (dnephin) [inline internal/server/models/group.go:16]: This field was not used, so removing it was as easy as removing this one line. We can always add it back in the future if we need it, but we'll have to add support to `CreateGroup` to actually use the column.

Right now most code seems to use the `Groups` field on `Identity`.

File: internal/server/data/group.go [hunk 2]
```diff
@@ -71,59 +93,49 @@ func groupIDsForUser(tx ReadTxn, userID uid.ID) ([]uid.ID, error) {
...
-		_, err := db.Exec("DELETE FROM identities_groups WHERE identity_id = ? AND group_id = ?", id, groupID)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
+// RemoveUsersFromGroup removes any user ID listed in idsToRemove from the group
+// with ID groupID.
+// Note that DeleteGroup also removes users from the group.
+func RemoveUsersFromGroup(tx WriteTxn, groupID uid.ID, idsToRemove []uid.ID) error {
```
Comment (dnephin) [inline internal/server/data/group.go:135]: This changed from a for loop performing one query per userID, to a single query using an `IN` clause. There will be some limit to the size of the query I guess, but fewer round trips to the database for most cases.

`DeleteGroup` has a similar query without the `identity_id IN` clause for removing all the members of a group.

File: internal/server/data/group.go [hunk 3]
```diff
@@ -71,59 +93,49 @@ func groupIDsForUser(tx ReadTxn, userID uid.ID) ([]uid.ID, error) {
...
 }
 
-func AddUsersToGroup(db GormTxn, groupID uid.ID, idsToAdd []uid.ID) error {
-	for _, id := range idsToAdd {
-		// This is effectively an "INSERT OR IGNORE" or "INSERT ... ON CONFLICT ... DO NOTHING" statement which
-		// works across both sqlite and postgres
-		_, err := db.Exec("INSERT INTO identities_groups (group_id, identity_id) SELECT ?, ? WHERE NOT EXISTS (SELECT 1 FROM identities_groups WHERE group_id = ? AND identity_id = ?)", groupID, id, groupID, id)
-		if err != nil {
-			return err
+func AddUsersToGroup(tx WriteTxn, groupID uid.ID, idsToAdd []uid.ID) error {
```
Comment (dnephin) [inline internal/server/data/group.go:117]: This changed from a for loop performing one query per userID, to a single query that inserts multiple entries. It also uses `ON CONFLICT DO NOTHING` now that we only need to support postgreSQL.

File: internal/server/data/group.go [hunk 4]
```diff
@@ -71,59 +93,49 @@ func groupIDsForUser(tx ReadTxn, userID uid.ID) ([]uid.ID, error) {
 	return result, rows.Err()
 }
 
-func DeleteGroups(db GormTxn, selectors ...SelectorFunc) error {
-	toDelete, err := ListGroups(db, nil, selectors...)
+func DeleteGroup(tx WriteTxn, id uid.ID) error {
```
Comment (dnephin) [inline internal/server/data/group.go:96]: This was only ever called with a single group ID, so this function has changed to only accept that one ID, which removes the need to list groups.

Group membership also changed from using `RemoveUsersFromGroup` to a query that deletes all rows that match the group ID.

**Review by pdevine — APPROVED** [review]
LGTM.

**Pull request merged**